### PR TITLE
fix verification of payProUrl equals null

### PR DESCRIPTION
**Pull request by greggzigler**
Now verifier.js (line 104) matches api.js (line 836) regarding payProUrl in signature

### CODE REVIEW

File: test/client.js [hunk 1]
```diff
@@ -1151,28 +1151,29 @@ describe('client API', function() {
...
+        amount: 20000,
+        toAddress: toAddress,
+        message: null,
+      }, {
+        amount: 30000,
+        toAddress: toAddress,
+      }]
+    };
+    function doit(opts, doNotVerifyPayPro, doBroadcast, done) {
       helpers.createAndJoinWallet(clients, 1, 1, function(w) {
```
Comment (isocolsky) [inline test/client.js]: Could you please move creation of the wallet & address outside of this function so it only gets called once at the start of the test?

File: lib/api.js [hunk 2]
```diff
@@ -903,7 +903,7 @@ API.prototype.sendTxProposal = function(opts, cb) {
     feePerKb: opts.feePerKb,
     payProUrl: opts.payProUrl,
     type: opts.type,
-    outputs: opts.outputs,
+    outputs: _.cloneDeep(opts.outputs),
```
Comment (troggy) [inline lib/api.js]: why is this needed?

File: lib/verifier.js [hunk 3]
```diff
@@ -101,7 +101,7 @@ Verifier.checkTxProposalBody = function(credentials, txp) {
     var proposalHeader = {
       outputs: outputs,
       message: txp.encryptedMessage || txp.message || null,
-      payProUrl: txp.payProUrl || undefined
+      payProUrl: txp.payProUrl
```
Comment (troggy) [inline lib/verifier.js]: this will produce different hashes for headers with ``null`` and ``undefined`` payProUrl. Does it make sense? Wouldn't it be better to make them equal (by nullifying undefined value)?

File: lib/api.js (hunk 2 above)
Comment (greggzigler) [inline lib/api.js]: because at line 910, the message is overwritten with the encrypted message. without cloning, the sendTxProposal() function will modify the client's copy of opts.

File: lib/verifier.js (hunk 3 above)
Comment (greggzigler) [inline lib/verifier.js]: when i experimented with replacing 'undefined' with 'null' here, sendTxProposal fails when the client passes in payProUrl = undefined. i think the root cause of the problem is that line 104 of verifier.js needs to match line 836 of api.js.

Comment (troggy) [inline lib/verifier.js]: yes, both verifier and api should have the same logic. I still think that undefined value for payProUrl should be nullified (in both places).

Btw perhaps you might be interested to follow on this: https://github.com/bitpay/bitcore-wallet-utils/pull/29
